OpenSim/Analyses/StaticOptimizationTarget.cpp

```diff
@@ -479,7 +479,8 @@ objectiveFunc(const Vector &parameters, const bool new_parameters, Real &perform
     int na = _model->getActuators().getSize();
     double p = 0.0;
     for(int i=0;i<na;i++) {
-        p +=  pow(fabs(parameters[i]),_activationExponent);
+        if (ScalarActuator* act = dynamic_cast<ScalarActuator*>(&_model->getActuators().get(i)))
```

Comment:
This seems like the wrong time to be doing the dynamic_cast. The `parameters`  should already be correctly sized to (size == na) in which case there is no reason to perform this check. Also performing numerous dynamic_casts  in the heart of the optimization is going to degrade performance.